stdlib/socket/from_test.go

```diff
@@ -0,0 +1,241 @@
+,,0,1970-01-01T00:00:00Z,a,b,-0.3,true
+,,0,1970-01-01T00:00:00Z,a,b,10.0,true
+,,0,1970-01-01T00:00:00Z,a,b,5.33,true
+,,1,1970-01-01T00:00:00Z,b,b,0.42,true
+,,1,1970-01-01T00:00:00Z,b,b,0.1,true
+,,1,1970-01-01T00:00:00Z,b,b,-0.3,true
+,,1,1970-01-01T00:00:00Z,b,b,10.0,true
+,,1,1970-01-01T00:00:00Z,b,b,5.33,true
+,,2,1970-01-01T00:00:00Z,b,b,0.42,false
+,,2,1970-01-01T00:00:00Z,b,b,0.1,false
+,,2,1970-01-01T00:00:00Z,b,b,-0.3,false
+,,2,1970-01-01T00:00:00Z,b,b,10.0,false
+,,2,1970-01-01T00:00:00Z,b,b,5.33,false
+`,
+			want: []*executetest.Table{
+				{
+					KeyCols: []string{"tag1", "tag2", "boolean"},
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "tag1", Type: flux.TString},
+						{Label: "tag2", Type: flux.TString},
+						{Label: "double", Type: flux.TFloat},
+						{Label: "boolean", Type: flux.TBool},
+					},
+					Data: [][]interface{}{
+						{execute.Time(0), "a", "b", 0.42, true},
+						{execute.Time(0), "a", "b", 0.1, true},
+						{execute.Time(0), "a", "b", -0.3, true},
+						{execute.Time(0), "a", "b", 10.0, true},
+						{execute.Time(0), "a", "b", 5.33, true},
+					},
+				},
+				{
+					KeyCols: []string{"tag1", "tag2", "boolean"},
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "tag1", Type: flux.TString},
+						{Label: "tag2", Type: flux.TString},
+						{Label: "double", Type: flux.TFloat},
+						{Label: "boolean", Type: flux.TBool},
+					},
+					Data: [][]interface{}{
+						{execute.Time(0), "b", "b", 0.42, true},
+						{execute.Time(0), "b", "b", 0.1, true},
+						{execute.Time(0), "b", "b", -0.3, true},
+						{execute.Time(0), "b", "b", 10.0, true},
+						{execute.Time(0), "b", "b", 5.33, true},
+					},
+				},
+				{
+					KeyCols: []string{"tag1", "tag2", "boolean"},
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "tag1", Type: flux.TString},
+						{Label: "tag2", Type: flux.TString},
+						{Label: "double", Type: flux.TFloat},
+						{Label: "boolean", Type: flux.TBool},
+					},
+					Data: [][]interface{}{
+						{execute.Time(0), "b", "b", 0.42, false},
+						{execute.Time(0), "b", "b", 0.1, false},
+						{execute.Time(0), "b", "b", -0.3, false},
+						{execute.Time(0), "b", "b", 10.0, false},
+						{execute.Time(0), "b", "b", 5.33, false},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			id := executetest.RandomDatasetID()
+			d := executetest.NewDataset(id)
+			c := execute.NewTableBuilderCache(executetest.UnlimitedAllocator)
+			c.SetTriggerSpec(flux.DefaultTrigger)
+			r := ioutil.NopCloser(bytes.NewReader([]byte(tc.input)))
+			ss, err := socket.NewSocketSource(tc.spec, r, id, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// Add `yield` in order to add `from` output tables to cache.
+			ss.AddTransformation(executetest.NewYieldTransformation(d, c))
```

Comment:
Ah I see the reason for the yield now. Disregard my previous comment.

Reply:
OK, if `yieldTransformation` is only for tests then lets move it to a test package.

Reply:
I see its in the executetest package we are good then.